versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoDBAliasCollectorImpl.java

```diff
@@ -43,16 +46,16 @@ public String escape(String unescaped) {
   public String alias(Entity value) {
     switch (value.getType()) {
       case MAP:
-        throw new UnsupportedOperationException("Unable to convert: " + value.getType());
+        return "{ $map {" + value.getMap().entrySet().stream().map(e -> String.format("%s: %s", e.getKey(), e.getValue()))
+          .collect(Collectors.joining(", ")) + "}}";
       case LIST:
-        throw new UnsupportedOperationException("Unable to convert: " + value.getType());
+        return "[" + value.getList().stream().map(Entity::toString).collect(Collectors.joining(", ")) + "]";
       case NUMBER:
         return String.valueOf(value.getNumber());
       case STRING:
         return value.getString();
       case BINARY:
-        // TODO investigate if this will be required.
-        throw new UnsupportedOperationException("Unable to convert: " + value.getType());
+        return Base64.getEncoder().encodeToString(value.getBinary().toByteArray());
```

Comment:
I don't think this is sufficient to represent binary in Mongo.

Reply:
I'm thinking create a BsonBinary object then use toString to convert it back?

Reply:
What is the intent, to convert this to Mongo binary format? If so, please look up the correct format (I believe there are tests which do this in the codebase).

Reply:
The intent is to convert to a String representation of binary. I now use BsonBinary.